packages/react-native/ReactCommon/react/nativemodule/core/platform/ios/ReactCommon/RCTTurboModuleManager.mm

```diff
@@ -808,9 +808,26 @@ - (Class)_getModuleClassFromName:(const char *)moduleName
     moduleClass = [_delegate getModuleClassFromName:moduleName];
   }
 
-  if (!moduleClass) {
-    moduleClass = getFallbackClassFromName(moduleName);
+  // Early return to avoid unnecessary computation
+  if (moduleClass != nil) {
+    return moduleClass;
+  }
+
+  moduleClass = getFallbackClassFromName(moduleName);
+  if (moduleClass != nil) {
+    return moduleClass;
   }
+
+  // fallback on modules registered throught RCT_EXPORT_MODULE with custom names
+  NSString *objcModuleName = [NSString stringWithUTF8String:moduleName];
+  auto modules = RCTGetModuleClasses();
+  for (Class current in modules) {
+    NSString *currentModuleName = [current moduleName];
```

Comment:
ughhh i hate that we have to expose the possibility of selector not found crash but checking for the selector feels like overkill

Reply:
Yeah, I thought the same, but I checked the code that put modules in the `moduleClasses` array and the check is done [at insertion time](https://github.com/facebook/react-native/blob/main/packages/react-native/React/Base/RCTBridge.m#L52-L60). So, it felt safe not to check it here.

Reply:
BTW, I think it's safe to refactor the `RCTModuleClasses` variable like this:
```diff
- static NSMutableArray<Class> *RCTModuleClasses;
+ static NSMutableArray<id<RCTBridgeModule>> *RCTModuleClasses;
```

It will require a bit of work, but we can make the whole system much more type-safe.